threescale/types.go

```diff
@@ -154,6 +144,16 @@ type UsageReport struct {
 // UsageReports defines a map of metric names to 'UsageReport'
 type UsageReports map[string]UsageReport
 
+// Transaction holds the params and optional additions that will be sent
+// to 3scale as query parameters or headers.
+type Transaction struct {
+	Metrics    Metrics
+	Params     Params
+	Timestamp  string
+	context    context.Context
+	extensions Extensions
```

Comment:
These are currently just per request and it's unlikely they will ever be per transaction because it would end up wasting lots of space (I think you change this in another commit, though).

Reply:
I don't fix it actually but had thought about it. So I wanted a way to surface some obscure functionality to the end user. I also wanted to avoid bloating the API for simple use cases, but maybe we need to accept `*Options` in the interface methods, wdyt?

Reply:
Yes, some mechanism to specify these per request which you don't need to bother (not too much anyway) if you are not using it.